typing/typecore.ml

```diff
@@ -2026,7 +2060,7 @@ and type_pat_aux
           pat_desc = Tpat_alias(q, id, name);
           pat_loc = loc; pat_extra=[];
           pat_type = q.pat_type;
-          pat_mode = q.pat_mode;
+          pat_mode = mode_cross !env expected_ty q.pat_mode;
```

Comment:
I'm a little suspicious of this line. I'd expect `pat_mode` (the mode in the Typedtree pattern) to match the mode passed to `enter_variable` (the mode of the binding in the environment), like in the `Ppat_var` case. Should this call to `mode_cross` come before `enter_variable`?